orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py

```diff
@@ -550,6 +582,71 @@ def test_changes_input_requires_grad_reinitializes_module_gradient_graph_builder
     model(x)
     assert module_gradient_graph_builder != model._module_gradient_graph_builder
 
+@pytest.mark.parametrize("device", ['cuda'])
+def test_input_requires_grad_backward_creates_input_grad_as_required0(device):
+    N, D_in, H, D_out = 32, 784, 500, 10
+    model = NeuralNetMultiplePositionalArgumentsMultipleOutputs0(D_in, H, D_out).to(device)
+    model = ORTModule(model)
+    x1 = torch.randn(N, D_in, device=device, requires_grad=True)
+    x2 = torch.randn(N, D_in, device=device, requires_grad=True)
+
+    y1, _ = model(x1, x2)
+    s1 = y1.sum()
+    s1.backward()
+    assert x1.grad is not None and x2.grad is not None
+
+    # named_params[0] and named_params[1] correspond to weight and bias for fc1, similarly
+    # named_params[2] and named_params[3] correspond to weight and bias for fc2.
+    named_params = list(model.named_parameters())
+    assert torch.count_nonzero(named_params[0][1].grad) > 0
+    assert torch.count_nonzero(named_params[1][1].grad) > 0
+    assert named_params[2][1].grad is None or torch.count_nonzero(named_params[2][1].grad) == 0
+    assert named_params[3][1].grad is None or torch.count_nonzero(named_params[3][1].grad) == 0
+
+    # Reset gradients
+    for param in named_params:
+        param[1].grad = None
+
+    _, y2 = model(x1,x2)
+    s2 = y2.sum()
+    s2.backward()
+    named_params = list(model.named_parameters())
+    assert named_params[0][1].grad is None or torch.count_nonzero(named_params[0][1].grad) == 0
+    assert named_params[1][1].grad is None or torch.count_nonzero(named_params[1][1].grad) == 0
+    assert torch.count_nonzero(named_params[2][1].grad) > 0
+    assert torch.count_nonzero(named_params[3][1].grad) > 0
+
+
+
+@pytest.mark.parametrize("x1_requires_grad, x2_requires_grad", [(True, True), (True, False), (False, False), (False, True)])
+def test_input_requires_grad_backward_creates_input_grad_as_required1(x1_requires_grad, x2_requires_grad):
+
+    def run_step(model, x1, x2):
+        y1, y2 = model(x1, x2)
+        s = y2.sum()
+        s.backward()
+
+    N, D_in, H, D_out = 32, 784, 500, 10
+    device = 'cuda'
+    pt_model = NeuralNetMultiplePositionalArgumentsMultipleOutputs1(D_in, H, D_out).to(device)
+    ort_model = ORTModule(pt_model)
```

Comment:
nit. 
we should create a deepcopy of the pt_model here, otherwise ort_model and pt_model are sharing the same underlaying weights. 

it's ok for this test case, because we are not updating the weight values. If we have another test case that modify the weight values in run_step(), then you will see unmatching results.